pkg/cmd/auth/status/status.go

```diff
@@ -180,18 +187,31 @@ func statusRun(opts *StatusOptions) error {
 	stdout := opts.IO.Out
 	cs := opts.IO.ColorScheme()
 
-	statuses := make(map[string]Entries)
+	showToken := opts.ShowToken
+	if opts.Exporter != nil && slices.Contains(opts.Exporter.Fields(), "token") {
+		showToken = true
+	}
+
+	statuses := make(map[string][]authEntry)
 
 	hostnames := authCfg.Hosts()
 	if len(hostnames) == 0 {
 		fmt.Fprintf(stderr,
 			"You are not logged into any GitHub hosts. To log in, run: %s\n", cs.Bold("gh auth login"))
+		if opts.Exporter != nil {
+			opts.Exporter.Write(opts.IO, struct{}{})
+			return nil
+		}
```

Comment:
Let's add a comment here that in machine-friendly mode, we always exit with zero.


```suggestion
		if opts.Exporter != nil {
			// In machine-friendly mode, we always exit with no error. 
			opts.Exporter.Write(opts.IO, struct{}{})
			return nil
		}
```